File: C++/1.자료구조/queue&stack/스택10828.cpp
// #include <vector>
// #include <algorithm>
// #include <iostream>
// #include <stack>
// using namespace std;

// int main()
// {
//     stack<int> st;
//     int t;
//     cin >> t;
//     while(t--)
//     {
//         string a;
//         cin >> a;
//         if(a=="push")
//         {
//             int b;
//             cin >> b;
//             st.push(b);
//         }
//         else if(a=="pop")
//         {
//             if(!st.empty())
//             {
//                 int x = st.top();
//                 st.pop();
//                 cout << x << '\n';
//             }
//             else cout << -1 << '\n';
//         }
//         else if(a=="size")
//         {
//             cout << st.size() << '\n';
//         }
//         else if(a=="empty")
//         {
//             cout << st.empty() << '\n';
//         }
//         else if(a=="top")
//         {
//             if(!st.empty())
//             {
//                 cout << st.top() << '\n';
//             }
//             else cout << -1 << '\n';
//         }
//     }
//     return 0;
// }
#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int n;
    cin >> n;
    stack<int> st;
    for(int i=0;i<n;i++) {
      string x;
      cin >> x;
      if(x=="push") {
        int y;
        cin >> y;
        st.push(y);
      }
      else if(x=="pop") {
        if(st.empty()) cout << -1 << '\n';
        else {
          cout << st.top() << '\n';
          st.pop();
        }
      }
      else if(x=="size") {
        cout << st.size() << '\n';
      }
      else if(x=="empty") {
        if(st.empty()) cout << 1 << '\n';
        else cout << 0 << '\n';
      }
      else if(x=="top") {
        if(st.empty()) cout << -1 << '\n';
        else cout << st.top() << '\n';
      }
    }
    return 0;
}

File: Java/1.자료구조/queue&stack/스택10828.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		StringTokenizer st;
		Stack<Integer> stack = new Stack<Integer>();


		int n = Integer.parseInt(br.readLine());
		for(int i=0;i<n;i++) {
			st = new StringTokenizer(br.readLine()," ");
			String x = st.nextToken();
			if(x.equals("push")) {
				int y = Integer.parseInt(st.nextToken());
				stack.push(y);
			} else if(x.equals("pop")) {
				if(stack.empty()) sb.append(-1).append("\n");
				else {
					sb.append(stack.peek()).append("\n");
					stack.pop();
				}
			} else if(x.equals("size")) {
				sb.append(stack.size()).append("\n");
			} else if(x.equals("empty")) {
				if(stack.empty()) sb.append(1).append("\n");
				else sb.append(0).append("\n");
			} else if(x.equals("top")) {
				if(stack.empty()) sb.append(-1).append("\n");
				else sb.append(stack.peek()).append("\n");
			}
		}
		System.out.println(sb);
	}
}
